File: common/errors/dotnet/Sentry.cs
using Sentry;

namespace Common.Errors;

public static class SentryHelper
{
    /// <summary>
    /// Initialize Sentry if SENTRY_DSN is set
    /// </summary>
    public static void InitSentry(string serviceName)
    {
        var dsn = Environment.GetEnvironmentVariable("SENTRY_DSN");
        if (string.IsNullOrEmpty(dsn))
        {
            return; // Silently skip if no DSN
        }

        try
        {
            SentrySdk.Init(options =>
            {
                options.Dsn = dsn;
                options.Environment = Environment.GetEnvironmentVariable("APP_ENV") ?? "dev";
                options.Release = Environment.GetEnvironmentVariable("GIT_SHA");
                options.TracesSampleRate = 1.0;
            });

            SentrySdk.ConfigureScope(scope =>
            {
                scope.SetTag("service", serviceName);
            });
        }
        catch
        {
            // Silently fail if Sentry not available
        }
    }

    /// <summary>
    /// Capture exception with service context
    /// </summary>
    public static void CaptureException(Exception exception, string serviceName, Dictionary<string, object>? context = null)
    {
        try
        {
            SentrySdk.ConfigureScope(scope =>
            {
                scope.SetTag("service", serviceName);
                if (context != null)
                {
                    scope.SetContext("additional", context);
                }
            });
            SentrySdk.CaptureException(exception);
        }
        catch
        {
            // Silently fail
        }
    }
}

File: common/errors/js/sentry.ts
/**
 * Initialize Sentry for JavaScript/TypeScript services (Frontend, API Gateway)
 * Only initializes if SENTRY_DSN or NEXT_PUBLIC_SENTRY_DSN is set
 */
export function initSentry(serviceName: string): void {
  const dsn = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN;
  
  if (!dsn) {
    return; // Silently skip if no DSN
  }
  
  try {
    // Dynamic import to avoid errors if @sentry/nextjs or @sentry/node not installed
    if (typeof window !== 'undefined') {
      // Browser/Next.js client-side
      import('@sentry/nextjs').then((Sentry) => {
        Sentry.init({
          dsn,
          environment: process.env.APP_ENV || process.env.NODE_ENV || 'dev',
          release: process.env.GIT_SHA,
          tracesSampleRate: 1.0,
        });
        Sentry.setTag('service', serviceName);
      }).catch(() => {
        // Sentry SDK not installed, silently continue
      });
    } else {
      // Node.js (API Gateway)
      import('@sentry/node').then((Sentry) => {
        Sentry.init({
          dsn,
          environment: process.env.APP_ENV || process.env.NODE_ENV || 'dev',
          release: process.env.GIT_SHA,
          tracesSampleRate: 1.0,
        });
        Sentry.setTag('service', serviceName);
      }).catch(() => {
        // Sentry SDK not installed, silently continue
      });
    }
  } catch (error) {
    // Silently fail if Sentry not available
  }
}

/**
 * Capture exception with service context
 */
export function captureException(error: Error, serviceName: string, context?: Record<string, any>): void {
  try {
    if (typeof window !== 'undefined') {
      import('@sentry/nextjs').then((Sentry) => {
        Sentry.setTag('service', serviceName);
        if (context) {
          Sentry.setContext('additional', context);
        }
        Sentry.captureException(error);
      }).catch(() => {});
    } else {
      import('@sentry/node').then((Sentry) => {
        Sentry.setTag('service', serviceName);
        if (context) {
          Sentry.setContext('additional', context);
        }
        Sentry.captureException(error);
      }).catch(() => {});
    }
  } catch {
    // Silently fail
  }
}

